src/Core/src/Hosting/Dispatching/AppHostBuilderExtensions.cs

```diff
@@ -10,25 +10,35 @@ public static partial class AppHostBuilderExtensions
 	{
 		public static MauiAppBuilder ConfigureDispatching(this MauiAppBuilder builder)
 		{
+			// register the DispatcherProvider as a singleton for the entire app
 			builder.Services.TryAddSingleton<IDispatcherProvider>(svc =>
 				// the DispatcherProvider might have already been initialized, so ensure that we are grabbing the
 				// Current and putting it in the DI container.
 				DispatcherProvider.Current);
 
-			builder.Services.TryAddScoped(svc =>
+			// register the Dispatcher as a singleton for the entire app as it may be different to secondary windows
+			builder.Services.TryAddSingleton(GetDispatcherFromDispatcherProvider);
+			// register the initializer so we can get the dispatcher in the application thread
+			builder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<IMauiInitializeService, DispatcherInitializer>());
+
+			// register the Dispatcher as a scoped service as each window may have different dispatchers
+			builder.Services.TryAddScoped(GetDispatcherFromDispatcherProvider);
+			// register the initializer so we can get the dispatcher in the window thread
+			builder.Services.TryAddEnumerable(ServiceDescriptor.Scoped<IMauiInitializeScopedService, DispatcherInitializer>());
```

Comment:
This currently crashes because `IMauiInitializeScopedService` is a lie

<img width="1315" alt="image" src="https://github.com/dotnet/maui/assets/5375137/7a9f36ac-18c6-4e52-8623-566cbed57df4">

https://github.com/dotnet/maui/issues/19591

Reply:
Rewritten an seems to run. Waiting for CI to confirm.